### PR TITLE
Update xxhash_64 to utilize the cuco equivalent implementation

### PR DESCRIPTION
**Pull request by PointKernel**
## Description
This PR updates `xxhash_64` to leverage the cuco equivalent implementation and makes it available as a cudf detail utility, as requested by @mhaseeb123. Additionally, it fixes a minor issue in the `murmurhash3_x64_128.cuh` header by adding the missing `#pragma once` directive.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cudf/hashing/detail/xxhash_64.cuh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/strings/string_view.cuh>
```
Comment (mhaseeb123) [inline cpp/include/cudf/hashing/detail/xxhash_64.cuh:19]: To include `normalize_nans()`
```suggestion
#include "hash_functions.cuh"

#include <cudf/strings/string_view.cuh>
```

**Review by mhaseeb123 — APPROVED** [review]
Thanks for quickly adding these changes.

File: cpp/include/cudf/hashing/detail/xxhash_64.cuh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "hash_functions.cuh"
```
[the remark this answers is not in the document]
Comment (ttnghia) [reply cpp/include/cudf/hashing/detail/xxhash_64.cuh:19]: Is there any other places that still need `hash_function.cuh`?

File: cpp/src/hash/xxhash_64.cu [hunk 3]
```diff
@@ -18,6 +18,7 @@
 #include <cudf/detail/utilities/algorithm.cuh>
 #include <cudf/hashing/detail/hash_functions.cuh>
```
Comment (mhaseeb123) [inline cpp/src/hash/xxhash_64.cu:19]: Maybe we can remove this header now as it was needed for `normalize_nans` which has now moved to the header file.

File: cpp/include/cudf/hashing/detail/xxhash_64.cuh (hunk 2 above)
[the remark this answers is not in the document]
Comment (PointKernel) [reply cpp/include/cudf/hashing/detail/xxhash_64.cuh:19]: I missed this header in the first place and @mhaseeb123 pointed out it's needed by `normalize_nans` (https://github.com/rapidsai/cudf/pull/17393/files#r1851262698)

File: cpp/src/hash/xxhash_64.cu (hunk 3 above)
Comment (PointKernel) [reply cpp/src/hash/xxhash_64.cu]: Good point. Removed

**Review by ttnghia — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

**Pull request merged**